lib/Jsrt/JsrtDebuggerObject.cpp

```diff
@@ -295,18 +289,13 @@ Js::DynamicObject * JsrtDebuggerStackFrame::GetLocalsObject(Js::ScriptContext* s
 
     Js::DynamicObject* globalsObject = nullptr;
 
-    if (this->pObjectModelWalker != nullptr)
-    {
-        HeapDelete(this->pObjectModelWalker);
-    }
-
     ReferencedArenaAdapter* pRefArena = scriptContext->GetThreadContext()->GetDebugManager()->GetDiagnosticArena();
     Js::IDiagObjectModelDisplay* pLocalsDisplay = Anew(pRefArena->Arena(), Js::LocalsDisplay, this->stackFrame);
-    this->pObjectModelWalker = pLocalsDisplay->CreateWalker();
+    WeakArenaReference<Js::IDiagObjectModelWalkerBase>* objectModelWalker = pLocalsDisplay->CreateWalker();
 
-    if (this->pObjectModelWalker != nullptr)
+    if (objectModelWalker != nullptr)
     {
```

Comment:
Only in case of OOM, no harm in having the check.

---
In reply to: [77227996](https://github.com/Microsoft/ChakraCore/pull/1521#discussion_r77227996) [](ancestors = 77227996)